---
title: "Data Inspection for Final Project"
output: github_document
date: '2022-04-20'
---
First, I will load in the packages (tidyverse, lemon, knitr). This code is in the script but not knit to the markdown document for readability. 
```{r,setup, include=FALSE}
library(tidyverse)
library(lemon)
library(knitr)
library(janitor)
library(dplyr )
library(ggplot2)
library(reshape2)
library(bold)
library(spider)
library(data.table)
library(msa)
library(ggtree)
library(treeio)
library(phangorn)
library(seqRFLP)
library(ape)
library(adegenet)
install.packages("adegenet")
#if (!requireNamespace("BiocManager", quietly=TRUE))
#install.packages("BiocManager")
#BiocManager::install("msa")
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")


#BiocManager::install("ggtree")
```

I also read in the "DS-KROK4BDJ.txt" file, which is a tsv downloaded from our paper of choice and (I believe) contains all the data we will need to work with, including sample IDs, geographic information, and sequences. 
```{r,my-chunk, echo=FALSE}
bdj<-read_tsv("DS-KROK4BDJ.txt", show_col_types = FALSE )
column_names <- colnames(bdj)
view(bdj)
```
After reading in the file, I inspect the column dimensions
```{r}
dim(bdj)
```

There are 80 columns and 124 rows in this file. Next check out column names. 
```{r}
kable(column_names)
```
Let's look at the first 5 rows in this file. It's a large chunk, but very interesting. Scroll side to side to see the whole chunk. 
```{r}
kable(bdj[1:5, ])
```
From this point, we can pick out columns that we want to work with. For practice, I will select the phylum-subspecies taxID and name columns (14 columns total) and convert them to factor. After this (hidden) code chunk, the "bdj" object (our data) columns with these names will be converted from "numeric" or "character" to "factor". Since they are now factors, we can run other analyses on them and visualize how many of each, etc. 

```{r, chunk, echo=FALSE, include=FALSE}

bdj$phylum_taxID = as.factor(as.numeric(bdj$phylum_taxID))
bdj$class_taxID = as.factor(as.numeric(bdj$class_taxID))
bdj$order_taxID = as.factor(as.numeric(bdj$order_taxID))
bdj$family_taxID = as.factor(as.numeric(bdj$family_taxID))
bdj$genus_taxID = as.factor(as.numeric(bdj$genus_taxID))
bdj$species_taxID = as.factor(as.numeric(bdj$species_taxID))
bdj$subspecies_taxID = as.factor(as.numeric(bdj$subspecies_taxID))
bdj$phylum_name = as.factor(as.character(bdj$phylum_name))
bdj$class_name = as.factor(as.character(bdj$class_name))
bdj$order_name = as.factor(as.character(bdj$order_name))
bdj$family_name = as.factor(as.character(bdj$family_name))
bdj$genus_name = as.factor(as.character(bdj$genus_name))
bdj$species_name = as.factor(as.character(bdj$species_name))
bdj$subspecies_name = as.factor(as.character(bdj$subspecies_name))
```
(The code is here, trust me. Check the Rmd file if you want to see it)

Next check out the various family names in this file. 

(note: I'm just doing all this to inspect the file, we don't have to use these processes.)
```{r}
unique(bdj$family_name) %>% as.matrix(family_name) -> family_name_view
kable(family_name_view)

```

I'd like to visualize how many samples per family we have. In other words, how many "Therididae"? How many "Anapidae"?

I'll use ggplot to do this. 
```{r}
ggplot(bdj, aes(x=reorder(family_name, family_name, function(x)-length(x)),fill=family_name)) +
  geom_bar() +
  labs(title="Individuals per Family", 
         x="Family Name", y = "Count") +
  theme(axis.text.x = element_text(angle = 90))

```

That looks so cool and I want to do another one. This one will be genus. 

```{r}

ggplot(bdj, aes(x=reorder(genus_name, genus_name, function(x)-length(x)),fill=genus_name)) +
  geom_bar() +
  labs(title="Individuals per Genus", 
         x="Family Name", y = "Count") +
  theme(axis.text.x = element_text(angle = 90)) +
  theme(legend.position="none") 
```

Not very easy to read but the colors are colorful.
```{r}
phylo <- bdj
view(phylo)
```
Now we prepare to retrieve the BOLD sequence data. 

```{r}
phylo1 <- phylo%>%add_column(retrieved_seq = NA )%>%select(1:8,81,everything())
```

We have created the empty column where sequences from BOLD will be put. 
```{r}
slicey <- phylo1%>%slice(1:10)
```
Created slicey to practice on

```{r}
check2<- bold_seq(taxon = NULL, ids = slicey$processid, bin = slicey$bin_uri, container = NULL, institutions = NULL, researchers = NULL, geo = NULL, marker = "COI-5P", response = FALSE)
```

Here we matched the bin_uri information to the BOLD database, retrieved the sequence. And next we checked manually the first few entries against the NCBI database to double check the correct species was identified. We used process_id and bin_uri to define the gene and the species. 

check2 was the practice data. 
Next we need to read the check2 into a table or similar and extract the sequences. 
First we make the BOLD output file into a dataframe. This is a very wide dataframe with many columns. 

```{r}
str(check2)
sequence_df<- as.data.frame(check2)
```

We want to extract the columns for "id" and "sequence". 


```{r}
```


```{r}
as.data.frame(sequence_df, col.names = c("id", "sequence"))
```
Credit for below code: https://stackoverflow.com/questions/26973029/split-one-row-after-every-3rd-column-and-transport-those-3-columns-as-a-new-row

```{r}
seqlist<- as.data.frame(matrix(unlist(sequence_df, use.names = FALSE), ncol = 4, byrow= TRUE))
```

```{r}
seqlist1=seqlist
colnames(seqlist1) = c("id", "name", "gene", "sequence")

seqlist2 <- as.tibble(seqlist1) %>%select(1,4)
```

now seqlist2 contains the data we need to merge with our original file
```{r}

```
###################################
#New Thing!!##




```{r}
check2_1<- bold_seq(taxon = NULL, ids = bdj$processid, bin = NULL, container = NULL, institutions = NULL, researchers = NULL, geo = NULL, marker = "COI-5P", response = FALSE)
```

Here we matched the bin_uri information to the BOLD database, retrieved the sequence. And next we checked manually the first few entries against the NCBI database to double check the correct species was identified. We used process_id and bin_uri to define the gene and the species. 

check2 was the practice data. 
Next we need to read the check2 into a table or similar and extract the sequences. 
First we make the BOLD output file into a dataframe. This is a very wide dataframe with many columns. 

```{r}
str(check2_1)
sequence_df2<- as.data.frame(check2_1)
sequence_df2
```

We want to extract the columns for "id" and "sequence". 


Credit for below code: https://stackoverflow.com/questions/26973029/split-one-row-after-every-3rd-column-and-transport-those-3-columns-as-a-new-row


```{r}
seqlist2<- as.data.frame(matrix(unlist(sequence_df2, use.names = FALSE), ncol = 4, byrow= TRUE))
```

```{r}
seqlist3=seqlist2
seqlist3 <- as.tibble(seqlist2) %>%select(1,4)
colnames(seqlist3) = c("processid", "retrieved_seq")
```


now seqlist3 contains the data we need to merge with our original file. "retrieved_seq" is the column with our retrieved sequences.  
```{r}
sequence_merge_main <- merge(seqlist3, bdj, by = "processid")
```

Our retrieved sequences and the original "bdj" file are merged by the "processid" column into the "sequence_merge_main" dataframe. This is now the dataframe for the first steps of the tree. 
Next step is to make the fasta file for the tree. 
Credit below code: Seth Bannister https://stackoverflow.com/questions/23374100/convert-table-into-fasta-in-r
```{r}
names <- (seqlist3$processid)
sequences <- (seqlist3$retrieved_seq)
dfintermediate <- data.frame(names,sequences)
df1.fasta = dataframe2fas(dfintermediate, file = "df1.fasta")
```
df.fasta1 is our new fasta file. Now we will use "msa" to align the reads. 
```{r}
#fastatry <- system.file("df1.fasta", package = "msa"
mySequences <- readDNAStringSet("df1.fasta")
mySequences
```
Next align the sequences. 
```{r}
myFirstAlignment <- msa(mySequences)
myFirstAlignmentconvert <- myFirstAlignment
```
Looks perfect!

```{r}
#msaPrettyPrint(myFirstAlignment, output="pdf", showNames="none",
#showLogo="none", askForOverwrite=FALSE, verbose=FALSE)
```
```{r}
alignmentConvert <- msaConvert(myFirstAlignmentconvert, "seqinr::alignment")
alignConvert <- msaConvert(myFirstAlignmentconvert, "bios2mds::align")
```
The below code is credit to https://fuzzyatelin.github.io/bioanth-stats/module-24/module-24.html#step_1
```{r}
library(ape)
dna <- fasta2DNAbin("df1.fasta")
D <- dist.dna(dna, model = "TN93")
length(D)
#number of pairwise distances, computed as n(n-1)/2
```
The number of pairwise distances is 7626. 


```{r}
temp <- as.data.frame(as.matrix(D))
matrix <- table.paint(temp, cleg=0, clabel.row=.5, clabel.col=0.5)
```

